engine/src/main/java/org/pentaho/di/trans/steps/ldapinput/LDAPConnection.java

```
@@ -0,0 +1,771 @@
+   * Converts the GUID to a readable string format
+   *
+   * @param inArr
+   * @return the formatted GUID
+   */
+  private static String byteToHexEncode( byte[] inArr ) {
+    StringBuilder guid = new StringBuilder();
+    for ( int i = 0; i < inArr.length; i++ ) {
+      StringBuilder dblByte = new StringBuilder( Integer.toHexString( inArr[i] & 0xff ) );
+      if ( dblByte.length() == 1 ) {
+        guid.append( "0" );
+      }
+      guid.append( dblByte );
+    }
+    return guid.toString();
+  }
+
+  public RowMeta getFields( String searchBase ) throws KettleException {
+    RowMeta fields = new RowMeta();
+    List<String> fieldsl = new ArrayList<String>();
+    try {
+      search( searchBase, null, 0, null, SEARCH_SCOPE_SUBTREE_SCOPE );
+      Attributes attributes = null;
+      fieldsl = new ArrayList<String>();
+      while ( ( attributes = getAttributes() ) != null ) {
+
+        NamingEnumeration<? extends Attribute> ne = attributes.getAll();
+
+        while ( ne.hasMore() ) {
+          Attribute attr = ne.next();
+          String fieldName = attr.getID();
+          if ( !fieldsl.contains( fieldName ) ) {
+            fieldsl.add( fieldName );
+
+            String attributeValue = attr.get().toString();
+            int valueType;
+
+            // Try to determine the data type
+            //
+            if ( IsDate( attributeValue ) ) {
+              valueType = ValueMetaInterface.TYPE_DATE;
+            } else if ( IsInteger( attributeValue ) ) {
+              valueType = ValueMetaInterface.TYPE_INTEGER;
+            } else if ( IsNumber( attributeValue ) ) {
+              valueType = ValueMetaInterface.TYPE_NUMBER;
+            } else {
+              valueType = ValueMetaInterface.TYPE_STRING;
+            }
+
+            ValueMetaInterface value = ValueMetaFactory.createValueMeta( fieldName, valueType );
+            fields.addValueMeta( value );
+          }
+        }
+      }
+      return fields;
+    } catch ( Exception e ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)